component/loki/source/gcplog/internal/gcplogtarget/types.go

```diff
@@ -14,19 +15,23 @@ type Target interface {
 }
 
 // PullConfig configures a GCPLog target with the 'pull' strategy.
+// struct is a derived of 'GcplogTargetConfig', but specific for Pull configuration, from Promtail code (https://github.com/grafana/loki/blob/main/clients/pkg/promtail/scrapeconfig/scrapeconfig.go)
 type PullConfig struct {
-	ProjectID            string            `river:"project_id,attr"`
-	Subscription         string            `river:"subscription,attr"`
-	Labels               map[string]string `river:"labels,attr,optional"`
-	UseIncomingTimestamp bool              `river:"use_incoming_timestamp,attr,optional"`
+	ProjectID            string         `river:"project_id,attr"`
+	Subscription         string         `river:"subscription,attr"`
+	Labels               model.LabelSet `river:"labels,attr,optional"`
+	UseIncomingTimestamp bool           `river:"use_incoming_timestamp,attr,optional"`
+	UseFullLine          bool           `river:"use_full_line,attr,optional"`
 }
 
 // PushConfig configures a GCPLog target with the 'push' strategy.
+// struct is a derived of 'GcplogTargetConfig', but specific for Push configuration from Promtail code (https://github.com/grafana/loki/blob/main/clients/pkg/promtail/scrapeconfig/scrapeconfig.go)
 type PushConfig struct {
-	Server               *fnet.ServerConfig `river:",squash"`
-	PushTimeout          time.Duration      `river:"push_timeout,attr,optional"`
-	Labels               map[string]string  `river:"labels,attr,optional"`
-	UseIncomingTimestamp bool               `river:"use_incoming_timestamp,attr,optional"`
+	Server               server.Config  `river:",squash"`
```

Comment:
The river implementation needs to use the previous definition server. Let's revert this import change to _not_ use the weaveworks/common/server package instead.